File: Mobile/Projects/native/ClujEatsKotlin/app/src/main/java/en/ubb/clujeatskotlin/network/APIService.kt
package en.ubb.clujeatskotlin.network

import en.ubb.clujeatskotlin.db.Place
import io.reactivex.Observable
import retrofit2.Call
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.*

interface APIService {

    companion object {
        fun create(): APIService {
            val retrofit = Retrofit.Builder()
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .baseUrl("http://192.168.43.230:8000/")
                .build()
            return retrofit.create(APIService::class.java)
        }
    }

    @GET("places/")
    fun getAllPlaces() : Observable<List<Place>>

    @POST("places/new/")
    @Headers(
        "Content-Type: application/json"
    )
    fun addPlace(@Body place: Place) : Call<ResponseData>

    @DELETE("places/{id}/")
    fun deletePlace(@Path("id") id: Long) : Call<ResponseData>

    @PUT("places/{id}/")
    fun updatePlace(@Path("id") id: Long, @Body place: Place) : Call<ResponseData>
}

File: Mobile/Projects/non-native/ClujEatsFlutter/cluj_eats_flutter/lib/network/ApiService.dart
import 'dart:async';
import 'package:cluj_eats_flutter/domain/place.dart';
import 'package:http/http.dart' as http;

const baseUrl = "http://172.30.118.252:8000/";
const newUrl = 'places/new/';
const placesUrl = 'places/';

class APIService {

  static Future get(String url) {
    var path = baseUrl + url;
    return http.get(path);
  }

  static Future post(Place place){
    var placeMap =  place.toMap();
    var url = baseUrl + newUrl;
    return http.post(url, body: placeMap);
  }

  static Future put(Place place) {
    var placeMap =  place.toJson();
    var remoteId = place.remoteId;
    var url = baseUrl + placesUrl + remoteId.toString() + '/';
    return http.put(url, body: placeMap);
  }

  static Future delete(Place place) {
    var remoteId = place.remoteId;
    var url = baseUrl + placesUrl + remoteId.toString() + '/';
    return http.delete(url);
  }
}